testsuite/features/step_definitions/command_steps.rb

```diff
@@ -1456,29 +1456,33 @@
 end
 
 When(/^I change the server's short hostname from hosts and hostname files$/) do
-  old_hostname = get_target('server').hostname
+  server_node = get_target('server')
+  old_hostname = server_node.hostname
   new_hostname = old_hostname + '2'
   log "New short hostname: #{new_hostname}"
-
-  get_target('server').run("sed -i 's/#{old_hostname}/#{new_hostname}/g' /etc/hostname &&
-  echo '#{get_target('server').public_ip} #{get_target('server').full_hostname} #{old_hostname}' >> /etc/hosts &&
-  echo '#{get_target('server').public_ip} #{new_hostname}#{get_target('server').full_hostname.delete_prefix(get_target('server').hostname)} #{new_hostname}' >> /etc/hosts")
+  
+  server_node.run("sed -i 's/#{old_hostname}/#{new_hostname}/g' /etc/hostname &&
+  echo '#{server_node.public_ip} #{server_node.full_hostname} #{old_hostname}' >> /etc/hosts &&
+  echo '#{server_node.public_ip} #{new_hostname}#{server_node.full_hostname.delete_prefix(server_node.hostname)} #{new_hostname}' >> /etc/hosts")
+  get_target('server', refresh: true) # This will refresh the attributes of this node
 end
 
 # changing hostname
 When(/^I run spacewalk-hostname-rename command on the server$/) do
-  temp_server = twopence_init('server')
+  server_node = get_target('server')
   command = "spacecmd --nossl -q api api.getVersion -u admin -p admin; " \
-            "spacewalk-hostname-rename #{get_target('server').public_ip} " \
+            "spacewalk-hostname-rename #{server_node.public_ip} " \
             "--ssl-country=DE --ssl-state=Bayern --ssl-city=Nuremberg " \
             "--ssl-org=SUSE --ssl-orgunit=SUSE --ssl-email=galaxy-noise@suse.de " \
             "--ssl-ca-password=spacewalk"
-  out_spacewalk, result_code = temp_server.run(command, check_errors: false)
+  out_spacewalk, result_code = server_node.run(command, check_errors: false)
   log "#{out_spacewalk}"
-
+  
+  server_node = get_target('server', refresh: true) # This will refresh the attributes of this node
```

Comment:
If I understand this correctly, we only need such a refresh in these kinds of scenarios where we rename hostnames?

Reply:
Or any possible change in its attributes